src/edfi-paging-test/tests/api/test_request_client.py

```diff
@@ -0,0 +1,156 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import json
+import pytest
+
+import requests_mock
+
+from edfi_paging_test.api.request_client import RequestClient
+from http import HTTPStatus
+
+
+FAKE_KEY = "TEST_KEY"
+FAKE_SECRET = "TEST_SECRET"
+FAKE_ENDPOINT = "/ENDPOINT"
+API_BASE_URL = "https://localhost:54746"
+OAUTH_URL = "https://localhost:54746/oauth/token/"
+FAKE_API_RESPONSE_PAGE1 = [{"id" : "a"}, {"id" : "b"}]
+FAKE_API_RESPONSE_PAGE2 = [{"id" : "c"}, {"id" : "d"}]
+PAGE_SIZE = 2
+NUMBER_OF_PAGES = 2
+RETRY_COUNT = 0
+TOTAL_COUNT = 4
+TOKEN = "038f4cb947c04fb4851fc3792c6b004f"
+TOKEN_RESPONSE = {
+  "access_token": "038f4cb947c04fb4851fc3792c6b004f",
+  "expires_in": 1800,
+  "token_type": "bearer"
+}
+
+
+def describe_testing_RequestClient_class():
+    @pytest.fixture()
+    def default_request_client(mocker):
+        return RequestClient(api_base_url=API_BASE_URL, api_key=FAKE_KEY, api_secret=FAKE_SECRET, page_size=PAGE_SIZE, retry_count=RETRY_COUNT)
+
+    @pytest.fixture()
+    def paging_request_client(mocker):
+        request_client = RequestClient(api_base_url=API_BASE_URL, api_key=FAKE_KEY, api_secret=FAKE_SECRET, page_size=PAGE_SIZE, retry_count=RETRY_COUNT)
+        request_client._get_data = mocker.MagicMock(side_effect=[FAKE_API_RESPONSE_PAGE1, FAKE_API_RESPONSE_PAGE2, []])
+        return request_client
```

Comment:
You can chain fixtures together with pytest:
```suggestion
    def default_request_client(mocker):
        return RequestClient(api_base_url=API_BASE_URL, api_key=FAKE_KEY, api_secret=FAKE_SECRET, page_size=PAGE_SIZE, retry_count=RETRY_COUNT)

    @pytest.fixture()
    def paging_request_client(default_request_client, mocker):
        default_request_client._get_data = mocker.MagicMock(side_effect=[FAKE_API_RESPONSE_PAGE1, FAKE_API_RESPONSE_PAGE2, []])
        return default_request_client
```